lib/Common/DataStructures/WeakReferenceDictionary.h

```diff
@@ -44,4 +44,668 @@ namespace JsUtil
 
         PREVENT_COPY(WeakReferenceDictionary);
     };
+
+#if ENABLE_WEAK_REFERENCE_REGIONS
+
+    template <class TKey>
+    class WeakRefRegionValueDictionaryEntry : public SimpleDictionaryKeyEntry<TKey>
```

Comment:
Its probably worth adding a TODO here to unify this with basedictionary?

Reply:
I wasn't thinking of unifying with `BaseDictionary`, but instead having a new `BaseSplitWeakDictionary` or something where there is a list of `entries` and a separate list of recycler allocated things, which can be either keys or values. There's some complexity there since if the keys are weak references then lookups have to deal with keys nulling out.